src/main/groovy/com/pszymczyk/consul/ConsulStarter.groovy

```diff
@@ -29,9 +29,10 @@ class ConsulStarter {
     private AntUnzip unzip
 
     @PackageScope
-    ConsulStarter(Path dataDir, Path downloadDir, String consulVersion, File portsConfigFile, LogLevel logLevel, ConsulPorts ports) {
+    ConsulStarter(Path dataDir, Path downloadDir, Path configDir, String consulVersion, String customConfig, LogLevel logLevel, ConsulPorts ports) {
```

Comment:
I thinkg that it should be done by Consul internals :

> When loading configuration, Consul loads the configuration from files and directories in lexical order. For example, configuration file basic_config.json will be processed before extra_config.json. Configuration specified later will be merged into configuration specified earlier. In most cases, "merge" means that the later version will override the earlier. In some cases, such as event handlers, merging appends the handlers to the existing configuration. The exact merging behavior is specified for each option below.

https://www.consul.io/docs/agent/options.html